apps/shipments/iot_client.py

```diff
@@ -36,3 +42,94 @@ def update_shadow(self, device_id, shadow):
             raise AWSIoTError("Error in response from AWS IoT")
 
         return iot_shadow['data']
+
+    def get_list_owner_devices(self, owner_id, next_token=None, results=None, **kwargs):
+        LOG.debug(f'Getting devices for: {owner_id} from AWS IoT')
+        log_metric('transmission.info', tags={'method': 'DeviceAWSIoTClient.get_list_owner_devices'})
+
+        active = kwargs['active']
+        in_bbox = kwargs['in_bbox']
+
+        if in_bbox:
+            in_bbox = ','.join([c.strip() for c in in_bbox.split(',')])
+
+        if not next_token:
+            results = []
+
+        params_dict = OrderedDict(ownerId=owner_id,
```

Comment:
Does the ordering of parameters matter to the API Gateway? @rudnyg I know this is an internal endpoint but in general APIs shouldn't care about the order of query parameters

Reply:
Actually it doesn't. I set it this way in order to have consistency in the unittests since the IoT url are "hard coded", I didn't want to have a situation where a test fails because a mocked url is not found due to the params order.

Reply:
Ah I see. I know it's a pain, but could you refactor the unit test mocks to not care about the order of the query parameters? It could make adding unit tests in the future confusing if the query parameter order matters.

https://stackoverflow.com/a/9468284 may help - looks like a test.utils class that will parse URLs and compare them, including query parameters. I think you can use one of these `Url`s as the key for `iot_enpoints` in your unit tests and then use `Url` classes in the lookup, since the hashes should be the same, regardless of order

Reply:
Thanks for the hint :+1: